ruby_oop/mastermind/game.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+
+require './computer_player'
+require './human_player'
+
+# class to play the mastermind game
+class Game
+  attr_accessor :human_player, :computer_player
+
+  AVAILABLE_COLORS = ['🟤', '🔴', '🔵', '🟢', '🟡', '🟣'].freeze
+  @secret_code = []
+
+  def initialize
+    @human_player = codebreaker_or_codemaker
+    @computer_player = !@human_player
+  end
+
+  def computer_codemaker
+    puts "Computer is selecting random colors...\n\n"
+    @secret_code = AVAILABLE_COLORS.sample(4)
+    ComputerPlayer.new(@secret_code)
+  end
+
+  def computer_codebreaker
+    puts "Computer is selecting random colors...\n\n"
+    @guess_array = AVAILABLE_COLORS.sample(4)
+    ComputerPlayer.new(@guess_array)
+  end
+
+  def human_codemaker
+    puts 'Enter a code to be broken: '
+    @secret_code = [pick_color(color_choice_selection), pick_color(color_choice_selection),
+      pick_color(color_choice_selection), pick_color(color_choice_selection)]
+      HumanPlayer.new(@secret_code)
+  end
+
+  def human_codebreaker
+    @guess_array = [pick_color(color_choice_selection), pick_color(color_choice_selection),
+                    pick_color(color_choice_selection), pick_color(color_choice_selection)]
+    if @guess_array.include?(nil) || @guess_array.uniq.length != 4
+      puts 'Duplicate entries or empty entries not allowed. Enter a new selection again.'
+      @guess_array = [pick_color(color_choice_selection), pick_color(color_choice_selection),
+                      pick_color(color_choice_selection), pick_color(color_choice_selection)]
```

Comment:
más duplication here w/ @guess_array = [pick_color(color_choice_selection)...]. see comment above re: DRYing

Reply:
It looks like the right-side of that assignment `[pick_color(color_choice_selection)...]` is reused in `human_codemaker` too.  I would maybe look for a way to pull that out into its own method?